### PR TITLE
chore: update apply-mapping error message to be more verbose

### PR DESCRIPTION
**Pull request by johnbieren**

### CODE REVIEW

**Review by davidmogar — DISMISSED** [review]

**Review by happybhati — DISMISSED** [review]
Much needed change, thanks!

File: tasks/apply-mapping/apply-mapping.yaml [hunk 1]
```diff
@@ -202,7 +202,10 @@ spec:
 
         if [ "$(params.failOnEmptyResult)" = "true" ] && \
           [ "$(jq '.components | length' < "${SNAPSHOT_SPEC_FILE}")" -eq 0 ]; then
-          echo "ERROR: Resulting snapshot contains 0 components"
+          echo "ERROR: Resulting snapshot contains 0 components. This means that there were 0 components present in"
+          echo "both your Snapshot and your ReleasePlanAdmission mapping. Take a look at your component names and"
+          echo "make sure that all components from the snapshot you want to release are present in the"
```
Comment (mmalina) [inline tasks/apply-mapping/apply-mapping.yaml]: Heh, I am actually working on a similar PR :D But isn't this incorrect? You don't necessarily need to map all components. I guess you will in most cases, but I thought it was not mandatory.

Comment (davidmogar) [reply tasks/apply-mapping/apply-mapping.yaml]: It says all the components you want to release. I think it's ok.

Comment (davidmogar) [reply tasks/apply-mapping/apply-mapping.yaml]: Ok, you can read it the other way around I guess. Maybe `all the components you want to release are present in the `RPA`.

Comment (mmalina) [reply tasks/apply-mapping/apply-mapping.yaml]: Also, can we at least print the components for easy comparison:
```sh
echo "Components in snapshot: $(jq -c '[.components[].name]' < "${SNAPSHOT_SPEC_FILE_ORIG}")"
echo "Components in mapping: $(jq -c '[.components[].name]' < "${MAPPING}")"
```

Comment (johnbieren) [reply tasks/apply-mapping/apply-mapping.yaml]: I meant it as David read it "components you want to release."
I will modify the english and add the echo Martin suggested

Comment (johnbieren) [reply tasks/apply-mapping/apply-mapping.yaml]: updated

Comment (johnbieren) [reply tasks/apply-mapping/apply-mapping.yaml]: From the unit test logs:
```
[run-task : apply-mapping] ++ jq -c '[.components[].name]' /workspace/config/snapshot_spec.json.orig
[run-task : apply-mapping] Components in snapshot: ["comp1","comp2"]
[run-task : apply-mapping] + echo 'Components in snapshot: ["comp1","comp2"]'
[run-task : apply-mapping] ++ jq -c '[.components[].name]'
[run-task : apply-mapping] Components in mapping: ["comp3","comp4"]
[run-task : apply-mapping] + echo 'Components in mapping: ["comp3","comp4"]'
[run-task : apply-mapping] + exit 1
```

Comment (mmalina) [reply tasks/apply-mapping/apply-mapping.yaml]: Cool, thanks.

File: tasks/apply-mapping/README.md [hunk 2]
```diff
@@ -29,6 +29,9 @@ You can also expand image labels, e.g. "{{ labels.mylabel }}" -> The value of im
 | dataPath          | Path to the JSON string of the merged data to use in the data workspace                      | No       | -             |
 | failOnEmptyResult | Fail the task if the resulting snapshot contains zero components                             | Yes      | false         |
 
+## Changes in 1.7.1
```
Comment (mmalina) [inline tasks/apply-mapping/README.md]: Typo. 1.7.2

Comment (johnbieren) [reply tasks/apply-mapping/README.md]: fixed, thanks

**Review by mmalina — DISMISSED** [review]

**Review by happybhati — DISMISSED** [review]

**Comment by johnbieren** [comment]
@happybhati any shot I can have another approval since I had to rebase and resolve conflicts?

**Review by happybhati — APPROVED** [review]

**Pull request merged**